src/structural_properties.c

```diff
@@ -6006,36 +6006,50 @@ int igraph_i_avg_nearest_neighbor_degree_weighted(const igraph_t *graph,
  * Average nearest neighbor degree.
```

Comment:
I'm now actually starting to wonder why this is called `avg_`**`nearest`**`_neighbor_degree`. Shouldn't we actually change this to `avg_neighbor_degree`? "**Nearest** neighbor" seems to imply that we can also do something for neighbors that are further removed (in whatever sense), which we do not.